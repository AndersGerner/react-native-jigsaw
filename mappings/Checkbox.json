{"name":"Checkbox","tag":"Checkbox","category":"input","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Control_Checkbox.png","props":{"disabled":{"label":"Disabled","description":"Whether checkbox is disabled","editable":true,"required":false,"value":false,"type":"boolean"},"color":{"label":"Color","description":"Custom color for Checkbox","editable":true,"required":false,"type":"color"},"fieldName":{"label":"Field name","description":"The name of the field within the screen that will store this component's value","type":"fieldName","value":"checkboxValue","valuePropName":"status","handlerPropName":"onPress","editable":true,"required":false}},"layout":{}}