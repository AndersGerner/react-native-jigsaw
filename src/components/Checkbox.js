// @ts-nocheck
import * as React from "react";
import { Platform } from "react-native";
import CheckboxAndroid from "./CheckboxAndroid";
import CheckboxIOS from "./CheckboxIOS";
import { withTheme } from "../core/theming";

import {
  GROUPS,
  COMPONENT_TYPES,
  FORM_TYPES,
  FIELD_NAME,
} from "../core/component-types";

class Checkbox extends React.Component {
  // @component ./CheckboxAndroid.js
  static Android = CheckboxAndroid;

  // @component ./CheckboxIOS.js
  static IOS = CheckboxIOS;

  render() {
    return Platform.OS === "ios" ? (
      <CheckboxIOS {...this.props} />
    ) : (
      <CheckboxAndroid {...this.props} />
    );
  }
}

export default withTheme(Checkbox);

export const SEED_DATA = {
  name: "Checkbox",
  tag: "Checkbox",
  category: COMPONENT_TYPES.input,
  layout: null,
  props: {
    disabled: {
      group: GROUPS.data,
      label: "Disabled",
      description: "Whether checkbox is disabled",
      editable: true,
      required: false,
      defaultValue: false,
      formType: FORM_TYPES.boolean,
    },
    color: {
      group: GROUPS.basic,
      label: "Color",
      description: "Custom color for Checkbox",
      editable: true,
      required: false,
      formType: FORM_TYPES.color,
      defaultValue: null,
    },
    fieldName: {
      ...FIELD_NAME,
      defaultValue: "checkboxValue",
      valuePropName: "status",
      handlerPropName: "onPress",
    },
  },
};
