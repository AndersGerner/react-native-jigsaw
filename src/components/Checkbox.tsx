// @ts-nocheck
import * as React from "react";
import { Platform } from "react-native";
import CheckboxAndroid from "./CheckboxAndroid";
import CheckboxIOS from "./CheckboxIOS";
import { withTheme } from "../core/theming";

import {
  GROUPS,
  COMPONENT_TYPES,
  FORM_TYPES,
  PROP_TYPES,
  FIELD_NAME,
} from "../core/component-types";
import themeI from "../styles/DefaultTheme";

interface Props extends TouchableHighlightProps {
  status: "checked" | "indeterminate" | "unchecked";
  disabled: boolean;
  onPress: () => void;
  theme: typeof themeI;
  color?: string;
}

class Checkbox extends React.Component<Props> {
  // @component ./CheckboxAndroid.js
  static Android = CheckboxAndroid;

  // @component ./CheckboxIOS.js
  static IOS = CheckboxIOS;

  render() {
    return Platform.OS === "ios" ? (
      <CheckboxIOS {...this.props} />
    ) : (
      <CheckboxAndroid {...this.props} />
    );
  }
}

export default withTheme(Checkbox);

export const SEED_DATA = {
  name: "Checkbox",
  tag: "Checkbox",
  category: COMPONENT_TYPES.input,
  layout: null,
  props: {
    disabled: {
      group: GROUPS.data,
      label: "Disabled",
      description: "Whether checkbox is disabled",
      editable: true,
      required: false,
      defaultValue: false,
      formType: FORM_TYPES.boolean,
      propType: PROP_TYPES.BOOLEAN,
    },
    color: {
      group: GROUPS.basic,
      label: "Color",
      description: "Custom color for Checkbox",
      editable: true,
      required: false,
      formType: FORM_TYPES.color,
      propType: PROP_TYPES.THEME,
      defaultValue: null,
    },
    fieldName: {
      ...FIELD_NAME,
      defaultValue: "checkboxValue",
      valuePropName: "status",
      handlerPropName: "onPress",
    },
  },
};
