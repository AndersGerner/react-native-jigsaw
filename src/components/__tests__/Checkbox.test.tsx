import * as React from "react";
import renderer from "react-test-renderer";
import Checkbox from "../Checkbox";

it("renders checkbox", () => {
  const tree = renderer.create(<Checkbox />).toJSON();

  expect(tree).toMatchSnapshot();
});
