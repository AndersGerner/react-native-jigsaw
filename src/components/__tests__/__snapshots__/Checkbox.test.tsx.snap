// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checkbox 1`] = `
<View
  accessibilityLiveRegion="polite"
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(90, 69, 255, 1)",
      "borderColor": "rgba(165, 173, 183, 1)",
      "borderRadius": 2,
      "borderWidth": 0,
      "height": 25,
      "opacity": 1,
      "width": 25,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "opacity": 1,
        }
      }
    />
  </View>
</View>
`;
